text/0000-light-dom.md

```diff
@@ -0,0 +1,218 @@
+
+## Detailed design
+
+### Selecting Light DOM vs Shadow DOM
+
+In raw JS, developer has the choice to use or not use shadow DOM for their custom element. For example,
+
+```js
+// uses Shadow DOM
+class MyShadowElement extends HTMLElement {
+  connectedCallback() {
+    this.attachShadow({ mode: "open" });
+    this.shadowRoot.innerHTML = "<p>My custom element that uses Shadow DOM";
+  }
+}
+
+// doesn't use Shadow DOM
+class MyLightElement extends HTMLElement {
+  connectedCallback() {
+    this.innerHTML = "<p>My element that does not use Shadow DOM";
+  }
+}
+```
+
+In LWC as well, the developer will retain the choice: they can choose to inherit from `LightningElement` to select Shadow DOM (which already exists) or inherit from a new class, `MacroElement`, to opt into Light DOM.
+
+```js
+import { MacroElement } from "lwc";
+
+export default class MyLightComponent extends MacroElement {
+  renderedCallback() {
+    this.querySelector("child-element"); // note the absence of this.template
+  }
+}
+```
+
+`MacroElement` is a new class that will be exported from the `lwc` module. Its API and functionality remain identical to `LightningElement` with the primary difference of not using Shadow DOM.
+
+### Component features when using Light DOM
+
+Some of the LWC component capabilities are directly inherited from Shadow DOM, or emulated by the synthetic-shadow. Despite the use of Light DOM, we’d like to keep these features available, even if their behavior is slightly adapted to the Light DOM:
+
+#### Slots
+
+As we mentioned before, the component composition model in LWC is provided by slots. Light DOM will provide the same mental model for developers building Light DOM components.
+
+In Light DOM, `<slot>` will denote the place where the slotted component will be attached. The `<slot>` element itself won't be rendered. The slotted content (or the fallback content) will be flattened to the parent element at runtime.
+
+Since the `<slot>` element itself isn't rendered, adding attributes or event listeners to the `<slot>` element in the template will throw a compiler error.
+
+#### Styles
+
+Until now styles used in LWC components where scoped to the component thanks to shadow DOM (or synthetic shadow DOM) style scoping. In the light DOM, component styles are naturally leaks out of the component, LWC doesn't do any style scoping out of the box. Developers are in charge of making sure to write specific enough selector to target the component content.
+
+To support the cases where a shadow DOM element composes a light element, light DOM styles are required to be injected to the closest root node. For a given light DOM elements if all the ancestor components are also a light DOM components, the component style sheet will be injected in the document `<head>`. Otherwise if any of the ancestors is a shadow DOM components the style has to be injected in the closest shadow root. Upon insertion of a light DOM element does the following steps:
```

Comment:
```suggestion
To support the cases where a shadow DOM element composes a light element, light DOM styles are required to be injected to the closest root node. For a given light DOM elements if all the ancestor components are also a light DOM components, the component style sheet will be injected in the document `<head>`. Otherwise if any of the ancestors is a shadow DOM component, the style has to be injected in the closest shadow root. Upon insertion of a light DOM element, LWC does the following steps:
```